shiny/driver/mtldriver/window.go

```diff
@@ -124,7 +123,49 @@ func (w *windowImpl) Release() {
 	<-respCh
 }
 
-func (w *windowImpl) NextEvent() interface{} {
+func (w *Window) SetTitle(title string) error {
+	respCh := make(chan struct{})
+	w.chans.updateCh <- updateWindowReq{
+		window: w.window,
+		title:  &title,
+		respCh: respCh,
+	}
+	glfw.PostEmptyEvent() // Break main loop out of glfw.WaitEvents so it can receive on releaseWindowCh.
+	<-respCh
+	return nil
+}
+
+type attribPair struct {
+	key glfw.Hint
+	val int
+}
+
+func (w *Window) SetTopMost(topMost bool) error {
+	respCh := make(chan struct{})
+	val := glfw.True
+	if !topMost {
+		val = glfw.False
+	}
+	w.chans.updateCh <- updateWindowReq{
+		window: w.window,
+		attribs: []attribPair{{
+			key: glfw.Floating,
+			val: val,
+		}},
+		respCh: respCh,
+	}
+	glfw.PostEmptyEvent() // Break main loop out of glfw.WaitEvents so it can receive on releaseWindowCh.
+	<-respCh
+	return nil
+}
+
+// BUG: this doesn't work, and it doesn't error either
+func (w *Window) SetIcon(img image.Image) error {
+	w.window.SetIcon([]image.Image{img})
```

Comment:
TODO: the documentation suggests that you can call SetIcon from outside of the main thread, but maybe they are lying? Try sending this through the update window request channel.